sensor/kubernetes/listener/resources/convert.go

```diff
@@ -362,11 +373,41 @@ func (w *deploymentWrap) populateImageMetadata(localImages set.StringSet, pods .
 				image.Id = digest
 				image.NotPullable = !imageUtils.IsPullable(c.ImageID)
 				image.IsClusterLocal = localImages.Contains(image.GetName().GetFullName())
+				updateImageWithNewerImageName(image, runtimeImageName, false)
 			}
 		}
 	}
 }
 
+// updateImageWithNewerImageName will update the registry, remote, and full name
+// of image from a newer image name, such as the one provided by the container
+// runtime. This is needed in order to support CRI-O's unqualified search registries
+// and short name aliases, for more details refer to:
+// https://github.com/containers/image/blob/main/docs/containers-registries.conf.5.md
+//
+// The image in a workload's spec may differ from what the container runtime pulls,
+// therefore the image name must be updated to accurately reflect the pulled image
+// to give downstream processes (like scanning) a better chance at completing.
+func updateImageWithNewerImageName(image *storage.ContainerImage, newerImageName *storage.ImageName, updateDigest bool) {
+	if !features.UnqualifiedSearchRegistries.Enabled() || image.GetName() == nil || newerImageName == nil {
+		return
+	}
+
+	imageName := image.GetName()
+	origFullName := imageName.GetFullName()
+	if imageName.GetRegistry() != newerImageName.GetRegistry() || imageName.GetRemote() != newerImageName.GetRemote() {
+		imageName.Registry = newerImageName.GetRegistry()
+		imageName.Remote = newerImageName.GetRemote()
+
+		if updateDigest {
+			imageUtils.NormalizeImageFullName(imageName, image.GetId())
+		} else {
+			imageUtils.NormalizeImageFullNameNoSha(imageName)
+		}
```

Comment:
can you remind me why this part is necessary? Currently our full names remove the digest, right?

Reply:
Thats right, the image name object does not have a field for holding a digest (outside of the full name). 

Calling  `NormalizeImageFullNameNoSha` on image names that are referenced by only a digest (no tag) will yield an invalid full name, something like `quay.io/some/repo:`